pkg/block/block_test.go

```diff
@@ -205,11 +205,29 @@ func TestUpload(t *testing.T) {
 			{{Name: "b", Value: "1"}},
 		}, 100, 0, 1000, nil, 124, metadata.NoneFunc)
 		testutil.Ok(t, err)
-		err = Upload(ctx, log.NewNopLogger(), bkt, path.Join(tmpDir, b2.String()), metadata.NoneFunc)
+		err = Upload(ctx, log.NewNopLogger(), bkt, path.Join(tmpDir, b2.String()), metadata.NoneFunc, true)
 		testutil.NotOk(t, err)
 		testutil.Equals(t, "empty external labels are not allowed for Thanos block.", err.Error())
 		testutil.Equals(t, 4, len(bkt.Objects()))
 	}
+	{
+		// Upload block without debug meta files.
```

Comment:
Do we have a test that checks the `Upload block with debug meta files.` condition as well?

Reply:
Yes. There is one test case above this. https://github.com/thanos-io/thanos/blob/816628343d4de1a9b8f829178906b3387addd88c/pkg/block/block_test.go#L139

Reply:
That test is not verifying the presence of debug meta files, it should. Also, this test that you added, you should check that the debug meta files are NOT uploaded, you are not checking that.

Reply:
If we upload with debug meta files then 4 files are uploaded per block. So the test I wrote there are only 7 files in the obj. store (4 files from the test with debug meta files + 3 files from the test without debug metafiles).

Reply:
Gotcha :+1:  But I think, like we are verifying ChunkFiles, Meta File etc, we can explicitly verify debug metas too.

Reply:
Understood :) I will add that verification. 
I have 2 questions.
1. Should I add that verification in all test cases?
2. Is it good practice that one test case depends on another test case? Because the test case I wrote depends on the previous one. I decided to write this test case this way because otherwise, I have to make a new instance obj. store. WDYT? 🤗

Reply:
1. No need to add it in other tests, it'd be redundant.
2. In this case, I think it's not a good practice. What if someone adds a new test case in between?

Reply:
Got it 👍🏻 . I will change the test case accordingly. Thank you :)